### PR TITLE
[FIX] account_asset: company in moves generated by depreciation lines

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
**Description of the issue this PR addresses:**

- User in company A creates an account asset (the asset by default has company A).
- Change the company of the asset to company B.
- Fill the asset and Confirm it in order to have depreciation lines.
- Click on `Sell or Dispose` button. Result: the account moves created have company A instead of company B.

**Desired behavior after PR is merged:**

The account moves generated by clicking in `Sell or Dispose` button have the company of the asset instead of the company of the user.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by astirpe** [comment]
Any news about this?

**Comment by mreficent** [comment]
@qdp-odoo ping

**Comment by qdp-odoo** [comment]
looks the wrong patch to me: I think that's the company_id field of account.asset.asset that should be a related to the asset category, and so the company_id field of an asset should be readonly.

what do you think?

**Comment by mreficent** [comment]
What you propose doesn't sound bad, but that would imply changes to master instead of v11, right?

**Comment by qdp-odoo** [comment]
mmh not necessarily. One could argue that an asset with a different company_id than its category is an inconsistency in the DB and that should never happen. 

That's, at least, my point of view... Does @nim-odoo agree with me that we can target a stable version with my proposed change?

**Comment by qdp-odoo** [comment]
@nim-odoo pong

**Comment by nim-odoo** [comment]
What @qdp-odoo proposes seems to makes sense, indeed.

**Comment by mreficent** [comment]
@qdp-odoo the change is done. But this gives an error in [account_asset_demo_test.xml](https://github.com/odoo/odoo/blob/11.0/addons/account_asset/test/account_asset_demo_test.xml). Any idea in how to amend it?

**Comment by mreficent** [comment]
@qdp-odoo @nim-odoo It's ready :green_apple: :green_apple: :green_apple:

File: addons/account_asset/models/account_asset.py [hunk 1]
```diff
@@ -513,6 +512,7 @@ def create_move(self, post_move=True):
                 'date': depreciation_date or False,
                 'journal_id': category_id.journal_id.id,
                 'line_ids': [(0, 0, move_line_1), (0, 0, move_line_2)],
+                'company_id': category_id.company_id.id,
```
Comment (qdp-odoo) [inline addons/account_asset/models/account_asset.py]: not necessary

File: addons/account_asset/models/account_asset.py [hunk 2]
```diff
@@ -559,6 +559,7 @@ def create_grouped_move(self, post_move=True):
             'date': depreciation_date or False,
             'journal_id': category_id.journal_id.id,
             'line_ids': [(0, 0, move_line_1), (0, 0, move_line_2)],
+            'company_id': category_id.company_id.id,
```
Comment (qdp-odoo) [inline addons/account_asset/models/account_asset.py]: not necessary

**Comment by mreficent** [comment]
@qdp-odoo done!

**Comment by mreficent** [comment]
Ready! ✔️

**Comment by nim-odoo** [comment]
Reverted at https://github.com/odoo/odoo/commit/8ee2a0d1912e26dbd798acc0194581cca7a998f3 since it breaks existing databases.

**Comment by mreficent** [comment]
Or maybe it's easier if @KangOl doesn't forwardport this revert commit to `master` 🤗

**Pull request merged**